### PR TITLE
Add ability to associate EIPs with ENIs

### PR DESCRIPTION
**Pull request by evanccnyc**
Also disassociate

### CODE REVIEW

File: cloud/amazon/ec2_eip.py [hunk 1]
```diff
@@ -11,6 +11,10 @@
     description:
       - The EC2 instance id
     required: false
+  network_interface_id:
```
Comment (bcoca) [inline cloud/amazon/ec2_eip.py:14]: requires version_added

**Comment by evanccnyc** [comment]
Added the version added section.

**Comment by evanccnyc** [comment]
@bennojoy let me know if there is anything else I need to add.

**Comment by evanccnyc** [comment]
Any idea when this can be checked in?

**Comment by bcoca** [comment]
it is queued for testing, please have patience as our backlog is huge. Also this does not merge cleanly, it requires at least a rebase

**Comment by evanccnyc** [comment]
Gah it looks like a refactor was just checked in. Sigh. Okay Ill have to totally rewrite this.

**Comment by evanccnyc** [comment]
Closing this pull request. Submitted a new one with up to date code. #1871

**Pull request closed without merging**